src/com/owncloud/android/ui/errorhandling/ErrorMessageAdapter.java

```diff
@@ -126,6 +127,12 @@ public static String getResultMessage(final RemoteOperationResult result,
             return result.getData().get(0).toString();
         }
 
+        if(operation instanceof CreateFolderOperation && result.getHttpPhrase() != null && result.getHttpPhrase().length() > 0){
```

Comment:
It's possible that other 405 errors are triggered when creating a folder but not necessarily be related to a duplicated folder. To take into account all the existing 405 errors retrieved by the server, we could read the specific message from the response body as I suggested in https://github.com/owncloud/android/pull/2248#issuecomment-419012250. 

[Here](https://github.com/owncloud/android-library/blob/8e357d78119921afdf8bb3466c168b6165f480a0/src/com/owncloud/android/lib/common/operations/RemoteOperationResult.java#L263) you have some specific codes that we created in similar scenarios and for this case we could check whether `mHttpCode` is `HttpStatus.SC_METHOD_NOT_ALLOWED` and if so, parse the error message and set a new code like SPECIFIC_METHOD_NOT_ALLOWED.

If everything went well, the error message "The resource you tried to create already exists" should appear in the snackbar.

Reply:
@davigonz As you have suggested I could get the message from the CreateFolderOperation itself so that could be instead right? What I mean is that instead of checking `result.getHttpParse().length() > 0` I could instead check `result.getHttpParse().equals("The resource you tried to create already exists")` that would also work right?

Reply:
> What I mean is that instead of checking result.getHttpParse().length() > 0 I could instead check result.getHttpParse().equals("The resource you tried to create already exists") that would also work right?

You mean to set a more specific message than "The resource you tried to create already exists" with the name of the folder and so on. Think that using the message coming from the server is enough and we don't need to be aware about the changes in that server message, what could break the condition you said above.

Reply:
@davigonz I don't think the condition above would break as the output the response being returned is "The resource you tried to create already exists" checking for this response along with checking if the operation type was CreateFolderOperation should work.

Reply:
If server decides to change that message to _The resource you are trying to create already exists_ for example, what you suggested in https://github.com/owncloud/android/pull/2248#discussion_r223694676 would stop working. What I mean is that to make this error handling easier, we could just use the message retrieved from the server, following the steps I defined in https://github.com/owncloud/android/pull/2248#discussion_r223656714. 

Have a look at [this PR](https://github.com/owncloud/android-library/pull/166), in which we are reading antivirus message from response body, it might be helpful

Reply:
@davigonz Yes I tried this but for this to work I had to add 
```
case HttpConstants.HTTP_SERVICE_UNAVAILABLE:
                parseErrorMessageAndSetCode(
                        httpMethod.getResponseBodyAsString(),
                        ResultCode.SPECIFIC_SERVICE_UNAVAILABLE
                );
                break;
```
to the RemoteOperationResult.java file in the owncloud-android-library so that I could get the error message from the server response which can they be displayed on the snackbar directly  but I am unable to commit these changes.

Reply:
I can help you with this, you would need to follow the next steps:
1. Go to owncloud-android-library: `cd owncloud-android-library` from ownCloud Android app code, I guess you will be in branch `sd1998:Fix-2247`.
2. Create branch in owncloud-android-library
3. Add + commit + push changes
4. Go back to ownCloud Android app code (`cd ..`)
5. Update the library reference: `git add owncloud-android-library`
6. Commit + push changes

Reply:
@davigonz Thanks for the help. I have made the [PR](https://github.com/owncloud/android-library/pull/207) to the owncloud-android-library repo.